lib/tasks/gettext.rake

```diff
@@ -35,6 +35,31 @@ namespace :gettext do
         end
    end
 
+   desc 'Update locale files with slightly changed English msgids using a csv file of old to new strings'
+   task :update_msgids_from_csv do
+       mapping_file = find_mapping_file(ENV['MAPPING_FILE'])
+       mappings = {}
+       File.foreach(mapping_file) do |csv_line|
+           from,to = CSV.parse(csv_line.gsub('\"', '""')).first
```

Comment:
This looks like it would be easy to forget why the `gsub` is there.

I'd be inclined to extract the data cleaning in to `clean_csv_mapping_file`, and then parse the cleaned data:

```ruby
mappings = {}
CSV.parse(clean_csv_mapping_file(mapping_file)).each do |line|
  from, to = line
  mappings[from] = to
end

# Some descriptive comment about what the problem is…
def clean_csv_mapping_file(file)
  data = ''
  File.foreach(file) do |line|
    data += line.gsub('\"', '""')
  end
  data
end
```

Here's something you should be able to paste in to the console to verify the result is the same:

```ruby
require 'csv'
require 'tempfile'

MAPPING_CSV = <<-CSV
"{{reason}}, please sign in or make a new account.","{{reason}}, create an account or sign in"
"If you are thinking of using a pseudonym,\\n                please <a href=\\"{{url}}\\">read this first</a>.","<a href=\\"{{url}}\\">Thinking of using a pseudonym?</a>."
CSV

mapping_file = Tempfile.new('mappings.csv')
mapping_file.write(MAPPING_CSV)
mapping_file.close

## original

mappings = {}
File.foreach(mapping_file.path) do |csv_line|
  from,to = CSV.parse(csv_line.gsub('\"', '""')).first
  mappings[from] = to
end

mappings_original = mappings

## review

def clean_csv_mapping_file(file)
  data = ''
  File.foreach(file) do |line|
    data += line.gsub('\"', '""')
  end
  data
end

mappings = {}
CSV.parse(clean_csv_mapping_file(mapping_file.path)).each do |line|
  from, to = line
  mappings[from] = to
end

mappings == mappings_original

mapping_file.unlink
```